app/lib/dates.lib.js

```diff
@@ -9,6 +9,27 @@ const february = 2
 const lastDayOfFebruary = 28
 const lastDayOfFebruaryLeapYear = 29
 
+/**
+ * Given a return version and return cycle end date, return the earliest date as a string in the ISO format.
+ *
+ * @param {Array} returnVersion - The list of dates to check.
+ * @returns {String} - The earliest date as a string in the ISO format.
+ * @private
+ */
```

Comment:
This doesn't match what the function is actually doing or what the tests describe it as doing. Plus the JSDoc schema is incorrect.

```suggestion
/**
 * From an array of dates as strings, filter out empty values and return the earliest as a `Date`
 *
 * This was created as part of our work on generating return logs for licences, and needing to work out the earliest
 * end date between the return version end date, and the return cycle end date for the given year.
 *
 * @param {string[]} dates - The dates from which to select the earliest
 *
 * @returns {Date} The earliest date string as a `Date`
 */
```